lib/internal/util/comparisons.js

```diff
@@ -147,84 +202,114 @@ function looseDeepEqual(val1, val2, memos) {
   if (val2 === null || typeof val2 !== 'object') {
     return false;
   }
-  if (isDate(val1) && isDate(val2)) {
-    return val1.getTime() === val2.getTime();
-  }
-  if (isRegExp(val1) && isRegExp(val2)) {
-    return areSimilarRegExps(val1, val2);
-  }
-  if (val1 instanceof Error && val2 instanceof Error) {
-    if (val1.message !== val2.message || val1.name !== val2.name)
-      return false;
-  }
   const val1Tag = objectToString(val1);
   const val2Tag = objectToString(val2);
   if (val1Tag === val2Tag) {
-    if (!isObjectOrArrayTag(val1Tag) && isArrayBufferView(val1)) {
-      return areSimilarTypedArrays(val1, val2,
-                                   isFloatTypedArrayTag(val1Tag) ?
-                                     Infinity : 300);
+    if (isObjectOrArrayTag(val1Tag)) {
+      return keyCheck(val1, val2, kLoose, memos, kNoIterator);
+    }
+    if (isArrayBufferView(val1)) {
+      if (isFloatTypedArrayTag(val1Tag)) {
+        return areSimilarFloatArrays(val1, val2);
+      }
+      return areSimilarTypedArrays(val1, val2);
+    }
+    if (isDate(val1) && isDate(val2)) {
+      return getTime(val1) === getTime(val2);
```

Comment:
Ditto about getTime with a subclass (again +1 on the change, just making sure it's semver-major)